pinot-common/src/main/java/org/apache/pinot/common/utils/helix/HelixHelper.java

```diff
@@ -117,9 +124,11 @@ public Boolean call() {
           try {
             updatedIdealState = updater.apply(idealStateCopy);
           } catch (PermanentUpdaterException e) {
+            controllerMetrics.addMeteredValue(resourceName, ControllerMeter.IDEAL_STATE_UPDATE_FAILURE, 1L);
```

Comment:
We probably don't want to count this as failure because it will be retried. We should only count the catch clause over the `policy.attempt()` as failure.

Reply:
good call, updated